app/components/channel_info/index.js

```diff
@@ -0,0 +1,414 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    Dimensions,
+    Platform,
+    TouchableWithoutFeedback,
+    View,
+    Text,
+    findNodeHandle
+} from 'react-native';
+import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
+
+import EventEmitter from 'mattermost-redux/utils/event_emitter';
+
+import ErrorText from 'app/components/error_text';
+import FormattedText from 'app/components/formatted_text';
+import Loading from 'app/components/loading';
+import StatusBar from 'app/components/status_bar';
+import TextInputWithLocalizedPlaceholder from 'app/components/text_input_with_localized_placeholder';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import {getShortenedURL} from 'app/utils/url';
+
+export default class ChannelInfo extends PureComponent {
```

Comment:
Is there a better name we could use for this component? This name is confusing when there's already the ChannelInfo screen that doesn't involve this component. Maybe something like EditChannelInfo?